tests/integration/high_availability/test_self_healing.py

```diff
@@ -97,38 +97,41 @@ async def test_kill_db_process(
     logger.info("Ensuring all units have continuous writes incrementing")
     # ensure continuous writes still incrementing for all units
     async with ops_test.fast_forward():
-        await ensure_all_units_continuous_writes_incrementing(ops_test)
+        await ensure_all_units_continuous_writes_incrementing(ops_test, credentials=credentials)
 
     # ensure that we are able to insert data into the primary and have it replicated to all units
     database_name, table_name = "test-kill-db-process", "data"
-    await insert_data_into_mysql_and_validate_replication(ops_test, database_name, table_name)
-    await clean_up_database_and_table(ops_test, database_name, table_name)
+    await insert_data_into_mysql_and_validate_replication(
+        ops_test, database_name, table_name, credentials
+    )
+    await clean_up_database_and_table(ops_test, database_name, table_name, credentials)
 
 
 @pytest.mark.group(2)
 @pytest.mark.abort_on_fail
 @pytest.mark.unstable
 async def test_freeze_db_process(
-    ops_test: OpsTest, highly_available_cluster, continuous_writes
+    ops_test: OpsTest, highly_available_cluster, continuous_writes, credentials
 ) -> None:
     """Test to send a SIGSTOP to the primary db process and ensure that the cluster self heals."""
     mysql_application_name = get_application_name(ops_test, "mysql")
+    assert mysql_application_name, "mysql application name is not set"
 
     # ensure all units in the cluster are online
     assert await ensure_n_online_mysql_members(
         ops_test, 3
     ), "The deployed mysql application is not fully online"
 
     logger.info("Ensuring that all units continuous writes incrementing")
-    await ensure_all_units_continuous_writes_incrementing(ops_test)
+    await ensure_all_units_continuous_writes_incrementing(ops_test, credentials=credentials)
 
     mysql_unit = ops_test.model.applications[mysql_application_name].units[0]
     primary = await get_primary_unit(ops_test, mysql_unit, mysql_application_name)
 
     mysql_pid = await get_process_pid(
         ops_test, primary.name, MYSQL_CONTAINER_NAME, MYSQLD_PROCESS_NAME
     )
-    assert mysql_pid > 0, "mysql process id is not positive"
+    assert (mysql_pid or -1) > 0, "mysql process id is not positive"
```

Comment:
1) When can PID be missing?
2) below the error message is `is not known/positive` but here is just `not positive`. On purpose?

Reply:
this is just to quiet down the linter. the pid will be `None` when not found, causing linter messages on further use of the `mysql_pid` variable, which is very annoying